vividus-engine/src/test/java/org/vividus/BatchedPerformableTreeTests.java

```diff
@@ -89,6 +96,33 @@ void testPerformBeforeOrAfterStoriesPerformNoneForAfter()
         testPerformBeforeOrAfterStoriesPerformNone(Stage.AFTER);
     }
 
+    @Test
+    void shouldResolveIdenticalNamesBeforeStories()
+    {
+        PerformableRoot root = batchedPerformableTree.getRoot();
+        Story storyName = new Story("file:/tests/name.story");
+        Story storyFolderName = new Story("file:/tests/folder/name.story");
+        root.add(createPerformableStory(storyName));
+        root.add(createPerformableStory(storyFolderName));
+        batchedPerformableTree.performBeforeOrAfterStories(runContext, Stage.BEFORE);
+        assertEquals("folder" + File.separator + "name.story", storyFolderName.getName());
+    }
+
+    @Test
+    void shouldNotInteractWithStoriesAfterStories()
+    {
+        PerformableRoot root = batchedPerformableTree.getRoot();
+        Story storyMock = mock(Story.class);
+        PerformableStory performableStoryMock = mock(PerformableStory.class);
```

Comment:
```suggestion
        Story storyMock = mock();
        PerformableStory performableStoryMock = mock();
```
it can be simplified

Reply:
the same for similar cases